src/smif/decision/decision.py

```diff
@@ -349,131 +504,49 @@ def get_decision(self, results_handle):
         raise NotImplementedError
 
 
-class PreSpecified(DecisionModule):
-    """Pre-specified planning
-
-    Parameters
-    ----------
-    timesteps : list
-        A list of the timesteps included in the model horizon
-    register : dict
-        A dict of intervention dictionaries keyed by unique intervention name
-    planned_interventions : list
-        A list of dicts ``{'name': 'intervention_name', 'build_year': 2010}``
-        representing historical or planned interventions
+class RuleBased(DecisionModule):
+    """Rule-base decision modules
     """
-    def __init__(self, timesteps, register, planned_interventions):
-        super().__init__(timesteps, register)
-        self._planned = planned_interventions
 
-    def _get_next_decision_iteration(self):
-        return {
-            'decision_iterations': [0],
-            'timesteps': [x for x in self.timesteps]
-        }
-
-    def get_decision(self, results_handle):
-        """Return a dict of intervention names built in timestep
+    def __init__(self, timesteps, register):
+        super().__init__(timesteps, register)
+        self.satisfied = False  # type: bool
+        self.current_timestep = self.first_timestep  # type: int
+        self.current_iteration = 0  # type: int
+        # keep internal account of max iteration reached per timestep
+        self._max_iteration_by_timestep = {self.first_timestep: 0}
+        self.logger = getLogger(__name__)
 
-        Arguments
-        ---------
-        results_handle : smif.data_layer.data_handle.ResultsHandle
-            A reference to a smif results handle
+    def get_previous_iteration_timestep(self) -> Tuple[int, int]:
```

Comment:
Suggest making the return type `Optional[Tuple[int,int]]` and returning `None` rather than the empty tuple for the edge case where there's no previous iteration. mypy isn't happy that `tuple()` isn't `Tuple[int, int]`

- [ ] from typing import Optional 
- [ ] fix test_decision line 127

Reply:
Fixed in commit fb6e04